accounts-db/src/accounts_index.rs

```diff
@@ -1817,6 +1820,10 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> AccountsIndex<T, U> {
         reclaims: &mut SlotList<T>,
         max_clean_root_inclusive: Option<Slot>,
     ) {
+        if slot_list.len() <= 1 {
+            self.purge_older_root_entries_one_slot_list
```

Comment:
this has a high cost and should be avoided. We are marking the index entry as dirty no matter what below here. if this ever occurs again, we'd like to know.